Implement error scrubbing middelware with opts

LucasRoesler opened the pull request:
**What**
- Add new Option to scrub grpc error messages
- Allow the user to configure the scrubbing behavior using an optional
  function parameter
- Update the unit tests
- Add an "Err" method on the test proto server

**Notes**
- Had issues with vendoring and go 1.11 , which is why the aes package is copied in the way it is

LucasRoesler committed:
Implement error scrubbing middelware with opts

**What**
- Add new Option to scrub grpc error messages
- Allow the user to configure the scrubbing behavior using an optional
  function parameter
- Update the unit tests
- Add an "Err" method on the test proto server

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

trusch commented on grpc/errors.go:
```diff
@@ -0,0 +1,108 @@
+
+// WithErrorScrubbing ensures that internal or unknown errors do not leak information.
+// You  may pass a function to customize the scrubbing behavior. The package provides
+// two simple scrubbers: SimpleErrorScrubber and NoopErrorScrubber for convenience.
```
include encryting scrubber in comment

trusch commented on grpc/errors.go:
```diff
@@ -0,0 +1,108 @@
+
+// NewEncryptedErrorScrubber replaces unknown or internal errors with a generic error object. The
+// original error message is encrypted and added to the error message in the status.Details
+func NewEncryptedErrorScrubber(key string) func(error) error {
```
remove the `New` prefix to be uniform with the other two scrubbers

LucasRoesler replied:
But it is a scrubber factory

trusch commented on grpc/errors_test.go:
```diff
@@ -0,0 +1,70 @@
+package grpc
```
add a test for SimpleErrorScrubber and NewEncryptedErrorScrubber

LucasRoesler replied:
SimpleErrorScrubber is the default test

I will add one for the encrypted

LucasRoesler replied:
done

LucasRoesler committed:
Add test encrypted scrubbing, improve docs

**What**
- Add Decrypt method to aes
- Test the EncryptedErrorScrubber
- Update doc strings

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler commented:
@trusch i addressed each comment
